### PR TITLE
feat(web): polling indicator, manual refresh, stale-data warning

### LINKED ISSUE #65: feat(web): polling indicator, manual refresh, stale-data warning
## Context

`useGameStatus` polls every 20s but the UI is silent about it. Operators can't tell if data is current, the API is down, or the page is stale. There's no manual refresh button on most surfaces, and `useFileManager`'s reactive polling has the same problem.

## Scope

- New shared `<PollingIndicator>` component:
  - "Updated **3s** ago" label with a rotating refresh icon.
  - On hover, a tooltip showing next-poll countdown ("next refresh in 17s").
  - Fail state: when 2× the interval has passed without a successful response, show a red "stale" pill ("Stale · last updated 47s ago"). Auto-clears on next success.
- Promote the polling state out of `useGameStatus` into a small `<PollingProvider>` context so all routes can subscribe (Dashboard, Costs, Logs, Discord).
- Replace ad-hoc refresh buttons in components with a single Refresh button in the top bar that triggers all active polls.
- Wire the top-bar `LIVE` indicator (added in #59) to actual polling state — pulse cyan when fresh, dim when stale.

## Acceptance criteria

- [ ] `<PollingIndicator>` visible on every primary route.
- [ ] Stale state correctly enters at 2× interval without a successful response.
- [ ] Top-bar Refresh triggers all live polls and shows a brief loading affordance.
- [ ] No regressions in existing polling cadence (still 20s for status; reactive 5s while file-manager starting).

## Dependencies

- Part of #57.
- **Blocked by** #58, #59.

### PR DESCRIPTION
**Pull request by CoderCoco**
Closes #65

## Summary

Centralises polling state in a shared `PollingProvider` so every primary route can see when its data was last refreshed, when the next poll is due, and when it has gone stale (2× the interval without a successful response).

- New `polling/` module:
  - `PollingProvider` — registry of named pollers (interval + fn) with per-poller `lastSuccessAt` / `lastAttemptAt` / `loading` / `error` state, plus `refresh(name)` and `refreshAll()`. A 1Hz heartbeat drives "Xs ago" re-renders without each consumer running its own timer. Pure helper `isStale()` enforces the 2× rule.
  - `GameStatusProvider` — registers the dashboard's status + cost-estimate poll above the router so polling persists across `/logs`, `/costs`, `/discord`, `/settings`. Drop-in replacement for the old `useGameStatus` hook.
  - `PollingIndicator` — "Updated 3s ago" with a rotating refresh icon (spins while in-flight), a Radix tooltip for "next refresh in 17s", and a red "Stale · last updated 47s ago" pill that auto-clears on the next success.
- `useFileManager` registers/unregisters its reactive 5-second poll with the shared registry while the helper task is `starting`, so the LIVE indicator and top-bar Refresh see it.
- `AppLayout` top bar gets a Refresh button (icon spins while any poll is loading) and the LIVE dot now pulses cyan when fresh / dims when every poll is stale.
- `<PollingIndicator />` placed on every primary route header (Dashboard, Costs, Discord, Logs, Settings).

## Test plan

- [x] `npm run app:test` — 294/294 passing (new `PollingProvider.test.ts` covers `isStale` boundary cases; `LogsPage.test.tsx` wrapped in `PollingProvider`).
- [x] `npm run app:lint` — clean.
- [x] `npm run app:build` — clean (vite build).
- [x] `npx playwright test --list` — 53 specs enumerate, including the new `polling.spec.ts` (5 cases: "Updated …" visible on Dashboard / Logs / Discord / Settings, plus the top-bar Refresh button triggers an extra `GET /api/status`).
- [ ] Manual: open the dashboard, watch "Updated" count up over 20s, navigate between routes and confirm the indicator persists, click Refresh and confirm the icon spins.

---
_Generated by [Claude Code](https://claude.ai/code/session_01ASmrh4hn4qCM7unq7T3yk5)_

### CODE REVIEW

File: app/packages/web/src/polling/PollingProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  fnRef.current = fn;
+
+  useEffect(() => {
+    if (!enabled) return;
+    return ctx.register(name, () => fnRef.current(), intervalMs);
+  }, [ctx, name, intervalMs, enabled]);
+
+  return {
+    state: ctx.pollers[name],
+    refresh: () => ctx.refresh(name),
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingProvider.tsx]: Fixed in b965fc0. Pulled `register` and `refresh` out of the context up front so the effect's deps are the stable callbacks rather than the whole memoized value, which keeps the registration intact across the 1Hz heartbeat.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/hooks/useFileManager.ts [hunk 2]
```diff
@@ -50,19 +54,32 @@ export function useFileManager() {
...
-  useEffect(() => () => clearPoll(), []);
+  // Register the reactive poller only while the helper task is still spinning
+  // up. Unregisters automatically when the state transitions to running/stopped
+  // or the modal closes — that keeps the polling registry honest about which
+  // pollers are currently live.
+  useEffect(() => {
+    if (!activeGame || status?.state !== 'starting') return;
+    return ctx.register(
+      FILE_MANAGER_POLLER,
+      async () => {
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/hooks/useFileManager.ts:108]: Fixed in b965fc0 — same root cause as the `usePoller` thread. Destructured `register` out of the context so the effect only re-runs when `activeGame` / `status?.state` / `fetchOnce` change, keeping the 5-second cadence honest.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  return `${min}m ago`;
+}
+
+/** Build the "next refresh in X" tooltip body, falling back to "scheduled" while we wait for the first attempt. */
+function nextRefreshLabel(state: PollerState, now: number): string {
+  if (state.lastAttemptAt === null) return 'next refresh scheduled';
+  const elapsed = now - state.lastAttemptAt;
+  const remaining = Math.max(0, state.intervalMs - elapsed);
+  const sec = Math.round(remaining / 1000);
+  return `next refresh in ${sec}s`;
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingIndicator.tsx:96]: Fixed in a912b7f. `runOne()` now resets the poller's `setInterval` each time it fires (whether from the timer or from a manual `refresh` / `refreshAll`), so the next automatic poll lands `intervalMs` after the most recent attempt — which is exactly what the tooltip is rendering from `lastAttemptAt`. Manual Refresh now correctly reschedules to "next refresh in 20s".

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        if (!prev) return p;
+        return { ...p, [name]: { ...prev, loading: false, lastSuccessAt: Date.now(), error: null } };
+      });
+    } catch (err) {
+      setPollers((p) => {
+        const prev = p[name];
+        if (!prev) return p;
+        return { ...p, [name]: { ...prev, loading: false, error: err as Error } };
+      });
+    }
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingProvider.tsx:141]: Fixed in a912b7f. `runOne()` now bails early when the registry entry has `inFlight: true`, so a slow response, a parked 401 retry, or a manual Refresh during an in-flight call no longer stack additional fetches for the same poller. The flag is cleared in a `finally` block so a thrown poller still releases the slot.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/GameStatusProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ * the top-bar Refresh button can trigger it alongside any other active poll.
+ */
+export function GameStatusProvider({ children }: { children: ReactNode }) {
+  const [statuses, setStatuses] = useState<GameStatus[]>([]);
+  const [estimates, setEstimates] = useState<CostEstimates | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  const fetchAll = useCallback(async () => {
+    const [s, e] = await Promise.all([api.status(), api.costsEstimate()]);
+    setStatuses(s);
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/GameStatusProvider.tsx]: Fixed in a912b7f. `GameStatusProvider` now polls only `/api/status` on the 20-second cadence; `api.costsEstimate()` is fetched once on mount via a separate `useEffect`. The values only change on `terraform apply` and the dashboard's only consumer (KpiStrip / GameCard) doesn't need recurring updates, so this drops the steady ECS `getTaskDefinition` traffic on `/logs` / `/discord` / `/settings` while still populating the dashboard's estimates.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/GameStatusProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+ * Internally registers a `status` poller with the shared polling registry so
+ * the top-bar Refresh button can trigger it alongside any other active poll.
+ *
+ * Cost estimates are fetched once on mount rather than on every poll —
+ * `/api/costs/estimate` loops over every configured game and calls
+ * `EcsService.getTaskDefinition()` per game, so polling it every 20s on every
+ * route would add steady ECS API traffic for data that only changes on
+ * `terraform apply`.
+ */
+export function GameStatusProvider({ children }: { children: ReactNode }) {
```
Comment (CoderCoco) [inline app/packages/web/src/polling/GameStatusProvider.tsx:58]: Should have a unit test

File: app/packages/web/src/polling/PollingIndicator.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  className?: string;
+}
+
+/**
+ * "Updated 3s ago" label with a rotating refresh icon that spins while the
+ * poller is in-flight. On hover, shows the next-poll countdown. After 2× the
+ * interval without a success, switches to a red "Stale" pill that auto-clears
+ * on the next successful poll.
+ */
+export function PollingIndicator({ name = 'status', className }: Props) {
```
Comment (CoderCoco) [inline app/packages/web/src/polling/PollingIndicator.tsx:18]: Should have a unit test

File: app/packages/web/src/polling/PollingIndicator.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+      </Tooltip>
+    </TooltipProvider>
+  );
+}
+
+/**
+ * Format an absolute timestamp as "Xs ago" / "Xm ago" / "—" (when unset).
+ * We keep this sub-minute resolution since the poll interval is 20s.
+ */
+function formatAgo(ts: number | null, now: number): string {
```
Comment (CoderCoco) [inline app/packages/web/src/polling/PollingIndicator.tsx:82]: Unit test

File: app/packages/web/src/polling/PollingIndicator.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+  if (ts === null) return '—';
+  const sec = Math.max(0, Math.round((now - ts) / 1000));
+  if (sec < 60) return `${sec}s ago`;
+  const min = Math.round(sec / 60);
+  return `${min}m ago`;
+}
+
+/** Build the "next refresh in X" tooltip body, falling back to "scheduled" while we wait for the first attempt. */
+function nextRefreshLabel(state: PollerState, now: number): string {
```
Comment (CoderCoco) [inline app/packages/web/src/polling/PollingIndicator.tsx:91]: Unit test

File: app/packages/web/src/pages/CostsPage.tsx [hunk 10]
```diff
@@ -219,7 +220,10 @@ export function CostsPage() {
               ECS + Fargate spend, per-game estimates, and trailing-window deltas.
             </p>
           </div>
-          <RangeSelector active={range} onChange={setRange} />
+          <div className="flex items-center gap-3">
```
Comment (CoderCoco) [inline app/packages/web/src/pages/CostsPage.tsx:223]: Unit Test

File: app/packages/web/src/pages/DashboardPage.tsx [hunk 11]
```diff
@@ -41,17 +42,20 @@ export function DashboardPage() {
...
-          <Input
-            type="text"
-            value={query}
-            onChange={(e) => setQuery(e.target.value)}
-            placeholder="Filter by game or hostname…"
-            className="pl-9"
-            aria-label="Filter games"
-          />
+        {/* Search filter + polling indicator */}
+        <div className="mb-4 flex items-center justify-between gap-4">
```
Comment (CoderCoco) [inline app/packages/web/src/pages/DashboardPage.tsx:46]: Unit Test

File: app/packages/web/src/pages/DiscordPage.tsx [hunk 12]
```diff
@@ -145,7 +146,10 @@ export function DiscordPage() {
             permissions.
           </p>
         </div>
-        <ServerlessBadge cfg={cfg} />
+        <div className="flex items-center gap-3">
```
Comment (CoderCoco) [inline app/packages/web/src/pages/DiscordPage.tsx:154]: Unit Test

File: app/packages/web/src/components/AppLayout.tsx [hunk 13]
```diff
@@ -136,6 +136,56 @@ export function AppLayout({ children }: { children: ReactNode }) {
   );
 }
 
+/**
+ * Top-bar Refresh button — triggers every active poller in the registry. The
+ * icon spins while at least one poll is in flight so the operator gets a brief
+ * loading affordance even if the underlying call returns instantly.
+ */
+function RefreshAllButton() {
```
Comment (CoderCoco) [inline app/packages/web/src/components/AppLayout.tsx]: Unit Test

File: app/packages/web/src/components/AppLayout.tsx [hunk 14]
```diff
@@ -136,6 +136,56 @@ export function AppLayout({ children }: { children: ReactNode }) {
...
+    </Button>
+  );
+}
+
+/**
+ * Top-bar LIVE indicator — pulses cyan while at least one poller has a fresh
+ * success, dims gray when every poller is past 2× its interval, and goes
+ * neutral when no pollers are registered yet.
+ */
+function LiveIndicator() {
```
Comment (CoderCoco) [inline app/packages/web/src/components/AppLayout.tsx]: Unit Test

File: app/packages/web/src/polling/PollingProvider.tsx [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+
+  useEffect(() => {
+    if (!enabled) return;
+    return register(name, () => fnRef.current(), intervalMs);
+  }, [register, name, intervalMs, enabled]);
+
+  const refresh = useCallback(() => refreshCtx(name), [refreshCtx, name]);
+
+  return {
+    state: ctx.pollers[name],
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingProvider.tsx]: I agree. We should have a configurable polling time too to help reduce the queries.

[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingProvider.tsx]: Fixed in eeb8d72. Split `PollingContext` into two: `usePollingActions()` (stable `register` / `refresh` / `refreshAll`) and `usePollingState()` (the tick-driven `pollers` + `tick` slice). `usePoller` now subscribes to actions only, so `GameStatusProvider` and anything else that just registers a poller no longer re-renders on the 1Hz heartbeat — `BrowserRouter` and the route trees stay stable.

Also added a build-time override at `GAME_STATUS_INTERVAL_MS` via `import.meta.env.VITE_STATUS_POLL_MS` (per your follow-up about a configurable polling time) so deployments can lengthen the cadence without forking the source.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/hooks/useFileManager.ts [hunk 16]
```diff
@@ -50,19 +54,32 @@ export function useFileManager() {
...
+  }, [activeGame, fetchOnce]);
 
   const stop = useCallback(async () => {
     if (!activeGame) return;
     const result = await api.filesMgrStop(activeGame);
     setMessage(result.message);
-    pollRef.current = setTimeout(() => void poll(activeGame), 3000);
-  }, [activeGame, poll]);
+    void fetchOnce(activeGame);
+  }, [activeGame, fetchOnce]);
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/hooks/useFileManager.ts:85]: Fixed in eeb8d72. Added an explicit `pendingStop` flag that is set on a successful Stop call and cleared once `status.state` reaches `stopped` / `not_deployed`. The polling effect's gate is now `state === 'starting' || pendingStop`, so the registered poller stays alive across the post-Stop window when ECS is still reporting `running` and the modal updates as soon as the helper actually transitions.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/GameStatusProvider.tsx (hunk 6 above)
Comment (CoderCoco) [reply app/packages/web/src/polling/GameStatusProvider.tsx:58]: Done in eeb8d72 — `app/packages/web/src/polling/GameStatusProvider.test.tsx` covers status fetched on mount, cost estimates fetched exactly once (not per poll, addresses the related Copilot thread), and the consumer-outside-provider error path.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingIndicator.tsx (hunk 7 above)
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingIndicator.tsx:18]: Done in eeb8d72 — added `app/packages/web/src/polling/PollingIndicator.test.tsx` covering: the "Not polling" placeholder when no poller is registered, the "Updated …" label after a registered poller resolves, and the next-poll countdown surfacing through the Radix tooltip on hover.

The stale-pill branch is covered separately by the existing `isStale` unit tests in `PollingProvider.test.ts`, which exercise both the never-succeeded and the past-success paths.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingIndicator.tsx (hunk 8 above)
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingIndicator.tsx:82]: `formatAgo` is exercised indirectly by the "Updated …" assertion in `PollingIndicator.test.tsx` (eeb8d72). If you want a dedicated pure-function test for the "—" / `Xs ago` / `Xm ago` boundary cases, happy to follow up — just say the word and I'll either export it for testing or add a thin module that wraps it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingIndicator.tsx (hunk 9 above)
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingIndicator.tsx:91]: `nextRefreshLabel` is exercised end-to-end by the "next-poll countdown via tooltip on hover" assertion in `PollingIndicator.test.tsx` (eeb8d72). Same offer as the `formatAgo` thread — happy to add a dedicated pure-function test for the no-attempt-yet branch if you want it broken out.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/pages/CostsPage.tsx (hunk 10 above)
Comment (CoderCoco) [reply app/packages/web/src/pages/CostsPage.tsx:223]: The header-row indicator placement on Costs is currently covered at the e2e tier — the redesigned Costs page is heavy on Tooltip / Card / Table primitives and `useCostsData` is bound to `api.costsActual` / `api.costsEstimate`, so a unit test would need PollingProvider + GameStatusProvider + a TooltipProvider + the api mocks just to get the page to mount.

Instead I went with: (1) `PollingIndicator.test.tsx` covering the indicator behaviour directly, (2) `polling.spec.ts` (e2e) confirming "Updated …" appears on the Costs route after navigation. Happy to do a follow-up PR with a dedicated Costs page-level unit test if you'd prefer that broken out — say the word and I'll pick it up after this lands.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/pages/DashboardPage.tsx (hunk 11 above)
Comment (CoderCoco) [reply app/packages/web/src/pages/DashboardPage.tsx:46]: Same answer as the Costs thread — covered at the e2e tier (`polling.spec.ts` confirms the indicator renders on `/`) plus the dedicated `PollingIndicator.test.tsx`. Happy to add a Dashboard-page-level unit test as a follow-up if you want it broken out; it'd need PollingProvider + GameStatusProvider + BrowserRouter + the api mocks for `/api/status` / `/api/costs/estimate` / `/api/costs/actual` to mount, which is more setup than the indicator unit tests warranted on their own.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/pages/DiscordPage.tsx (hunk 12 above)
Comment (CoderCoco) [reply app/packages/web/src/pages/DiscordPage.tsx:154]: Same answer as the Costs and Dashboard threads — Discord page integration covered at e2e (`polling.spec.ts` navigates to `/discord` and asserts the indicator stays visible); the indicator's own behaviour is covered by `PollingIndicator.test.tsx`. Will pick up a dedicated DiscordPage page-level unit test in a follow-up if you want one — the page already needs PollingProvider + GameStatusProvider + the api mocks for `/api/discord/config` and `/api/games` just to mount.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/components/AppLayout.tsx (hunk 13 above)
Comment (CoderCoco) [reply app/packages/web/src/components/AppLayout.tsx]: Done in eeb8d72 — `app/packages/web/src/components/AppLayout.test.tsx` (`AppLayout — RefreshAllButton` describe block) covers the disabled-when-empty-registry state and that clicking the button fires every registered poller's fn.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/components/AppLayout.tsx (hunk 14 above)
Comment (CoderCoco) [reply app/packages/web/src/components/AppLayout.tsx]: Done in eeb8d72 — `app/packages/web/src/components/AppLayout.test.tsx` (`AppLayout — LiveIndicator` describe block) covers that the LIVE label always renders and that the dot picks up the cyan + `animate-pulse` classes once a registered poller has reported success.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/pages/CostsPage.tsx (hunk 10 above)
Comment (CoderCoco) [reply app/packages/web/src/pages/CostsPage.tsx:223]: I mean we could do that we just have to use mocks there. Can we ensure that we test this functionality using mocks and also keep the e2e tests. Update the claude.md with this pattern.

Comment (CoderCoco) [reply app/packages/web/src/pages/CostsPage.tsx:223]: Done in d6c3695.

- Added a shared `src/test-utils/renderPage.tsx` helper that wraps children in `PollingProvider → GameStatusProvider → MemoryRouter` so every page test mounts under the same provider stack the production app uses.
- `CostsPage.test.tsx`, `DashboardPage.test.tsx`, `DiscordPage.test.tsx` — each `vi.mock`s `../api.js` with `vi.hoisted`, returns canned data for the methods the page (and GameStatusProvider above it) calls, and asserts the page chrome + the `<PollingIndicator />` wiring resolves to "Updated …" once the mocked status poll completes.
- e2e specs untouched — the unit tier and the e2e tier now cover the indicator from two different angles.
- CLAUDE.md gets a new **"Routed page tests"** subsection under the existing "Two-tier browser testing strategy" describing exactly this pattern (renderPage helper, mocking the api client, intentionally complementary scope to e2e). Drops a checklist for "stub every method the page + GameStatusProvider calls" so future page tests don't hang waiting for the polling registry to settle.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/polling/PollingProvider.tsx [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+  return { refresh };
+}
+
+/** True iff `state` has gone longer than `STALE_MULTIPLIER × intervalMs` without a successful poll. */
+export function isStale(state: PollerState | undefined, now: number = Date.now()): boolean {
+  if (!state) return false;
+  if (state.lastSuccessAt === null) {
+    // Never succeeded — treat as stale once we've waited 2× the interval.
+    if (state.lastAttemptAt === null) return false;
+    return now - state.lastAttemptAt > STALE_MULTIPLIER * state.intervalMs;
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/polling/PollingProvider.tsx]: Fixed in 957a563. Added a `firstAttemptAt` field to `PollerState` that's set once on the first attempt and survives both retries and re-registrations. `isStale()` for the never-succeeded path now keys off `firstAttemptAt` instead of `lastAttemptAt`, so a poll that keeps failing every 20s correctly flips to stale once `2 × intervalMs` has elapsed since the first attempt. Added a regression test (`should ignore retry-driven lastAttemptAt updates so a perpetually-failing poll still flips stale`) that pins the new behaviour.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/hooks/useFileManager.ts [hunk 18]
```diff
@@ -1,68 +1,105 @@
...
     if (!activeGame) return;
     setMessage('Launching…');
     const result = await api.filesMgrStart(activeGame);
     setMessage(result.message);
-    if (result.success) {
-      pollRef.current = setTimeout(() => void poll(activeGame), 5000);
-    }
-  }, [activeGame, poll]);
+    if (result.success) void fetchOnce(activeGame);
+  }, [activeGame, fetchOnce]);
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/hooks/useFileManager.ts:77]: Fixed in 957a563. Added a symmetric `pendingStart` flag (alongside the existing `pendingStop`) so the polling effect's gate is now `state === 'starting' || pendingStart || pendingStop`. The flag is set on a successful Start and cleared once the status reaches `starting` / `running`. That covers the race where the first `/api/files/:game` read after `runTask` still says `stopped` because the new task isn't visible to `listTasksByStartedBy` yet — the poller stays registered until ECS catches up.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/packages/web/src/pages/DiscordPage.tsx [hunk 19]
```diff
@@ -145,7 +146,10 @@ export function DiscordPage() {
...
+          <PollingIndicator />
+          <ServerlessBadge cfg={cfg} />
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply app/packages/web/src/pages/DiscordPage.tsx:156]: Fixed in 957a563. The DiscordPage's pre-`cfg` branch now renders the same flex header as the loaded view (heading + description on the left, `<PollingIndicator />` on the right), so the indicator is visible across the full lifecycle: loading, error (`Discord config unavailable …`), and configured. Added `should keep the polling indicator visible while /api/discord/config is loading` in `DiscordPage.test.tsx` to pin it.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by CoderCoco** [comment]
@copilot take a final look at this and let me know if there are any super blocking bugs
